core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/GeoDistanceAggregationBuilder.java

```diff
@@ -167,25 +172,38 @@ private static Range parseRange(XContentParser parser) throws IOException {
         double from = 0.0;
         double to = Double.POSITIVE_INFINITY;
         String key = null;
-        String toOrFromOrKey = null;
+        String currentFieldName = null;
         Token token;
         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
-                toOrFromOrKey = parser.currentName();
+                currentFieldName = parser.currentName();
             } else if (token == XContentParser.Token.VALUE_NUMBER) {
-                if (Range.FROM_FIELD.match(toOrFromOrKey)) {
+                if (FROM_FIELD.match(currentFieldName)) {
                     from = parser.doubleValue();
-                } else if (Range.TO_FIELD.match(toOrFromOrKey)) {
+                } else if (TO_FIELD.match(currentFieldName)) {
                     to = parser.doubleValue();
+                } else {
+                    XContentParserUtils.throwUnknownField(currentFieldName, parser.getTokenLocation());
                 }
             } else if (token == XContentParser.Token.VALUE_STRING) {
-                if (Range.KEY_FIELD.match(toOrFromOrKey)) {
+                if (KEY_FIELD.match(currentFieldName)) {
                     key = parser.text();
-                } else if (Range.FROM_FIELD.match(toOrFromOrKey)) {
+                } else if (FROM_FIELD.match(currentFieldName)) {
                     fromAsStr = parser.text();
-                } else if (Range.TO_FIELD.match(toOrFromOrKey)) {
+                } else if (TO_FIELD.match(currentFieldName)) {
                     toAsStr = parser.text();
+                } else {
+                    XContentParserUtils.throwUnknownField(currentFieldName, parser.getTokenLocation());
+                }
+            } else if (token == XContentParser.Token.VALUE_NULL) {
+                if (FROM_FIELD.match(currentFieldName) || TO_FIELD.match(currentFieldName)
+                        || KEY_FIELD.match(currentFieldName)) {
+                    // ignore null value
+                } else if (KEY_FIELD.match(currentFieldName)) {
```

Comment:
is this condition not matched in the condition above? Maybe this should just be an else?